tensorflow/contrib/go/tensor_test.go

```diff
@@ -0,0 +1,630 @@
+package tensorflow_test
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	tf "github.com/tensorflow/tensorflow/tensorflow/contrib/go"
+)
+
+func getTensorFromGraph(t *testing.T, graphStr string) *tf.Tensor {
+	graph, err := tf.NewGraphFromText(graphStr)
+	if err != nil {
+		t.Fatal(err)
+	}
+	s, err := tf.NewSession()
+	if err := s.ExtendGraph(graph); err != nil {
+		t.Fatal(err)
+	}
+
+	output, err := s.Run(nil, []string{"output"}, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(output) != 1 {
+		t.Fatalf("The expexted number of tensors is 1 but there was %d tensors returned", len(output))
+		t.FailNow()
+	}
+
+	return output[0]
+}
+
+func TestStrDecode(t *testing.T) {
+	expectedResult := [][]byte{
+		[]byte("Hello1!"),
+		[]byte("Hello2!"),
+		[]byte("Hello3!"),
+	}
+	tensor := getTensorFromGraph(t, fmt.Sprintf(`
+		node {
+			name: "output"
+			op: "Const"
+			attr {
+				key: "dtype"
+				value {
+					type: DT_STRING
+				}
+			}
+			attr {
+				key: "value"
+				value {
+					tensor {
+						dtype: DT_STRING
+						tensor_shape {
+							dim {
+								size: 3
+							}
+						}
+						string_val: "%s"
+						string_val: "%s"
+						string_val: "%s"
+					}
+				}
+			}
+		}
+		version: 5`, string(expectedResult[0]), string(expectedResult[1]), string(expectedResult[2])),
+	)
+
+	result, err := tensor.AsStr()
+	if err != nil {
+		t.Error("Problem trying to cast a tensor into string slice, Error:", err)
```

Comment:
t.Fatal("Error casting tensor into string slice:", err)
(and eliminate FailNow) -- apply to rest. :)